homeassistant/components/nissan_leaf.py

```diff
@@ -0,0 +1,338 @@
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+# Currently waiting on the lib author to review a PR
+# So using my fork for now.
+# https://github.com/jdhorne/pycarwings2/pull/28
+
+REQUIREMENTS = ['https://github.com/BenWoodford/pycarwings2/archive/master.zip'
+                '#pycarwings']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'nissan_leaf'
+DATA_LEAF = 'nissan_leaf_data'
+
+DATA_BATTERY = 'battery'
+DATA_LOCATION = 'location'
+DATA_CHARGING = 'charging'
+DATA_PLUGGED_IN = 'plugged_in'
+DATA_CLIMATE = 'climate'
+DATA_RANGE_AC = 'range_ac_on'
+DATA_RANGE_AC_OFF = 'range_ac_off'
+
+CONF_NCONNECT = 'nissan_connect'
+CONF_INTERVAL = 'update_interval'
+CONF_CHARGING_INTERVAL = 'update_interval_charging'
+CONF_CLIMATE_INTERVAL = 'update_interval_climate'
+CONF_REGION = 'region'
+CONF_VALID_REGIONS = ['NNA', 'NE', 'NCI', 'NMA', 'NML']
+CONF_FORCE_MILES = 'force_miles'
+DEFAULT_INTERVAL = 30
+DEFAULT_CHARGING_INTERVAL = 15
+DEFAULT_CLIMATE_INTERVAL = 5
+
+CHECK_INTERVAL = 10
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_REGION): vol.In(CONF_VALID_REGIONS),
+        vol.Optional(CONF_NCONNECT, default=True): cv.boolean,
+        vol.Optional(CONF_INTERVAL, default=DEFAULT_INTERVAL): cv.positive_int,
+        vol.Optional(CONF_CHARGING_INTERVAL,
+                     default=DEFAULT_CHARGING_INTERVAL): cv.positive_int,
+        vol.Optional(CONF_CLIMATE_INTERVAL,
+                     default=DEFAULT_CLIMATE_INTERVAL): cv.positive_int,
+        vol.Optional(CONF_FORCE_MILES, default=False): cv.boolean
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+LEAF_COMPONENTS = [
+    'sensor', 'switch', 'binary_sensor'
+]
+
+SIGNAL_UPDATE_LEAF = 'nissan_leaf_update'
+
+
+def setup(hass, config):
+    username = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+
+    import pycarwings2
+
+    _LOGGER.debug("Logging into You+Nissan...")
+
+    try:
+        s = pycarwings2.Session(
+            username, password, config[DOMAIN][CONF_REGION])
+        _LOGGER.debug("Fetching Leaf Data")
+        leaf = s.get_leaf()
+    except(RuntimeError, urllib.error.HTTPError):
+        _LOGGER.error(
+            "Unable to connect to Nissan Connect with username and password")
+        return False
+    except KeyError:
+        _LOGGER.error(
+            "Unable to fetch car details..."
+            " do you actually have a Leaf connected to your account?")
+        return False
+    except:
+        _LOGGER.error(
+            "An unknown error occurred while connecting to Nissan: %s",
+            sys.exc_info()[0])
+
+    _LOGGER.info("Successfully logged in and fetched Leaf info")
+    _LOGGER.info(
+        "WARNING: This may poll your Leaf too often, and drain the 12V."

```

Comment:
We should not allow people to brick their car.

Reply:
This is mostly if they've left the car idle and on a low charge for a very long time. No helping that.